### PR TITLE
EB-20725: updated composer json to symfony 6

### PR DESCRIPTION
**Pull request by lucianduta**
### Updated loco adapter to symfony 6 and php 8.1
https://etsglobal.jira.com/browse/EB-20725 

### What has been done:
- updated Makefile containers to php8.1; 
- fixes for phpstan and unit tests; 
- removed symfony-bundle-test package (the last tag available at this moment is not compatible with php8)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -11,15 +11,15 @@
...
+        "symfony/yaml": "^4.4.20 || ^5.2.5 || ^6.0",
         "friendsofapi/localise.biz": "^1.0"
     },
     "require-dev": {
         "phpunit/phpunit": "^8.5",
-        "symfony/framework-bundle": "^4.4.20 || ^5.2.5",
+        "symfony/framework-bundle": "^4.4.20 || ^5.2.5 || ^6.0",
         "nyholm/psr7": "^1.2",
-        "nyholm/symfony-bundle-test": "^1.6.1",
+        "nyholm/symfony-bundle-test": "dev-master",
```
Comment (ketlane) [inline composer.json]: Using dev-master is not usually a good idea; can't we tie this dependency to a fixed version? (i.e. `^1.8.2`)

File: tests/Functional/BundleInitializationTest.php [hunk 2]
```diff
@@ -12,11 +12,10 @@
 namespace Translation\PlatformAdapter\Loco\Tests\Functional;
 
 use Http\HttplugBundle\HttplugBundle;
-use Nyholm\BundleTest\BaseBundleTestCase;
 use Translation\PlatformAdapter\Loco\Bridge\Symfony\TranslationAdapterLocoBundle;
 use Translation\PlatformAdapter\Loco\Loco;
 
-class BundleInitializationTest extends BaseBundleTestCase
+class BundleInitializationTest
```
Comment (ketlane) [inline tests/Functional/BundleInitializationTest.php:18]: Q: Why was this removed?

File: composer.json (hunk 1 above)
Comment (lucianduta) [reply composer.json]: I know, but the last tag available at this moment is not compatible with php8

File: tests/Functional/BundleInitializationTest.php (hunk 2 above)
Comment (lucianduta) [reply tests/Functional/BundleInitializationTest.php:18]: Because, apparently,  it does not exists anymore.

File: composer.json (hunk 1 above)
Comment (ketlane) [reply composer.json]: huh :(

File: tests/Functional/BundleInitializationTest.php (hunk 2 above)
Comment (ketlane) [reply tests/Functional/BundleInitializationTest.php:18]: Then is `nyholm/symfony-bundle-test` still necessary?

Comment (lucianduta) [reply tests/Functional/BundleInitializationTest.php:18]: I will try to remove it to check if it's really needed after I fix the other issues.

Comment (lucianduta) [reply tests/Functional/BundleInitializationTest.php:18]: I removed the package. It doesn't seem to be used anymore.

**Review by ketlane — APPROVED** [review]

**Review by ioana96 — APPROVED** [review]

File: composer.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "php-translation/loco-adapter",
+    "name": "etsglobal/loco-adapter",
```
Comment (ch3ric) [inline composer.json:2]: please keep the original package name so that we can propose a PR from this fork to the original repo (in order not to maintain our fork if this PR is accepted and merged in the original repo)

File: .github/workflows/static.yml [hunk 4]
```diff
@@ -8,7 +8,7 @@ jobs:
     steps:
       - uses: actions/checkout@master
       - name: Run PHPStan
-        uses: docker://jakzal/phpqa:php7.3-alpine
+        uses: docker://jakzal/phpqa:php8.1-alpine
```
Comment (ch3ric) [inline .github/workflows/static.yml:11]: then you also need to make sure the tests run on php 8.1 as well: https://github.com/ETSGlobal/loco-adapter/blob/55caa8cf04c1000db45743889bc5d0d0e2c0713c/.github/workflows/ci.yml#L13

File: tests/Functional/BundleInitializationTest.php (hunk 2 above)
Comment (ch3ric) [reply tests/Functional/BundleInitializationTest.php:18]: I don't understand why you say this bundle is not needed: this bundle https://github.com/SymfonyTest/symfony-bundle-test seems needed in case we want to test the kernel correctly boots whatever the symfony version with this bundle installed: this seems useful because this bundle supports all symfony version between Symfony 4 and 6.

Also this test is not failing currently because it's not a test anymore, because it's not extending PHPUnit TestCase... But if you extend this PHPUnit TestCase, you will see that it's failing because the method ->createKernel does not exist

**Review by ch3ric — COMMENTED** [review]
:writing_hand: @ketlane @lucianduta

File: tests/Functional/BundleInitializationTest.php (hunk 2 above)
Comment (ketlane) [reply tests/Functional/BundleInitializationTest.php:18]: The Symfony Test Bundle does not work with Symfony 6 https://github.com/SymfonyTest/symfony-bundle-test/blob/1.8.2/Readme.md -> we'd be forced to use the master branch, instead of a tag; and that's dangerous, for a library.

File: composer.json (hunk 3 above)
Comment (ketlane) [reply composer.json:2]: I think @lucianduta tried that, but he couldn't get the package to load, because composer will try to load the open-source `php-translations/loco-adapter` from Github, instead of loading our fork.

Comment (ch3ric) [reply composer.json:2]: It should work though, since it's what we're doing for other forks, for example for doctrine migration bundle: https://github.com/ETSGlobal/BUSINESS/blob/master/composer.json#L99 & https://github.com/ETSGlobal/BUSINESS/blob/master/composer.json#L18

Comment (ketlane) [reply composer.json:2]: Hm, maybe it only works when using a tag that does not exist in the base repo.

@lucianduta Here's a full tutorial https://medium.com/swlh/using-your-own-forks-with-composer-699358db05d9 that should work

**Comment by lucianduta** [comment]
CR was implemented on this PR https://github.com/ETSGlobal/loco-adapter/pull/2

File: composer.json (hunk 3 above)
Comment (lucianduta) [reply composer.json:2]: https://github.com/ETSGlobal/loco-adapter/pull/2

File: tests/Functional/BundleInitializationTest.php (hunk 2 above)
Comment (lucianduta) [reply tests/Functional/BundleInitializationTest.php:18]: https://github.com/ETSGlobal/loco-adapter/pull/2

Comment (ketlane) [reply tests/Functional/BundleInitializationTest.php:18]: Actually, since this is a library only used for tests (only required on the dev env), I guess it's not that dangerous to use the master branch.

**Comment by ch3ric** [comment]
fixed in https://github.com/ETSGlobal/loco-adapter/pull/2
:heavy_check_mark:
